File: Android/app/src/main/java/net/mrblindbandit/app/config/AppConfig.kt
package net.mrblindbandit.app.config

object AppConfig {
    const val APP_DISPLAY_NAME = "Mr. Blindbandit"
    const val MARKETING_VERSION = "1.7"
    const val API_BASE_URL = "https://api.mrblindbandit.net"

    const val PRIVACY_URL = "https://mrblindbandit.net/privacy/"
    const val TERMS_URL = "https://mrblindbandit.net/terms/"
    const val SUPPORT_URL = "https://mrblindbandit.net/support/"
    const val ACCESSIBILITY_URL = "https://mrblindbandit.net/accessibility/"
    const val ACCOUNT_DELETION_URL = "https://mrblindbandit.net/account/delete"
    const val SUPPORT_EMAIL = "business@mrblindbandit.net"

    /** Client-safe public configuration generated into BuildConfig. */
    val clerkPublishableKey: String get() = net.mrblindbandit.app.BuildConfig.CLERK_PUBLISHABLE_KEY
    val liveKitUrl: String get() = net.mrblindbandit.app.BuildConfig.LIVEKIT_URL
    val googleOAuthClientId: String get() = net.mrblindbandit.app.BuildConfig.GOOGLE_OAUTH_CLIENT_ID

    val isClerkConfigured: Boolean get() = clerkPublishableKey.startsWith("pk_")
    val isLiveKitConfigured: Boolean get() = liveKitUrl.startsWith("wss://")

    /** Phone OTP stays off until SMS/OTP is enabled in the production Clerk instance. */
    const val ENABLE_PHONE_OTP = false
}

File: App/AppConfig.swift
import Foundation

/// Central client configuration. Reusable backend secrets never ship in the binary.
enum AppConfig {
    static let appDisplayName = "Mr. Blindbandit"
    static let marketingVersion = "1.7"
    static let webBaseURL = URL(string: "https://mrblindbandit.net")!
    static let apiBaseURL = URL(string: "https://api.mrblindbandit.net")!
    static let defaultLiveKitRoom = "mrblindbandit"

    static let privacyURL = URL(string: "https://mrblindbandit.net/privacy/")!
    static let termsURL = URL(string: "https://mrblindbandit.net/terms/")!
    static let supportURL = URL(string: "https://mrblindbandit.net/support/")!
    static let accessibilityURL = URL(string: "https://mrblindbandit.net/accessibility/")!
    static let accountURL = URL(string: "https://mrblindbandit.net/account")!
    /// Public web page where anyone can request account deletion without the app (Google Play requirement).
    static let accountDeletionURL = URL(string: "https://mrblindbandit.net/account/delete")!
    static let supportEmail = "business@mrblindbandit.net"

    static var clerkPublishableKey: String { AppSecrets.clerkPublishableKey }
    static var liveKitURL: String { AppSecrets.liveKitURL }

    static var isClerkConfigured: Bool { clerkPublishableKey.hasPrefix("pk_") }

    /// Clerk production instance "mr. blindbandit - Blindbandit Records" (clerk.mrblindbandit.net).
    /// Email + password, Google, and Sign in with Apple. Apple is required by App Store Review
    /// Guideline 4.8 because Google sign-in is offered; the Apple provider must be enabled in Clerk.
    static var enableSignInWithApple: Bool { true }
}
